packages/object-prototype/test.js

```diff
@@ -0,0 +1,456 @@
+    Object.defineProperty(obj, 'bar', { enumerable: true });
+    assertExecution(t, 'propertyIsEnumerable', { obj, args: ['foo'], expected: false });
+    assertExecution(t, 'propertyIsEnumerable', { obj, args: ['bar'], expected: true });
+    assertExecution(t, 'propertyIsEnumerable', { obj, args: ['invalid'], expected: false });
+    assertNonExecutingFunctionPrototypeFunctions(t, 'propertyIsEnumerable', { obj });
+    t.end();
+  });
+
+  test('toLocaleString', function(t) {
+    const obj = generator();
+    assertFunction(t, 'toLocaleString', { obj, expected: obj.toString() });
+    t.end();
+  });
+
+  test('toString', function(t) {
+    assertFunction(t, 'toString', { generator, expected: '[object Object]' });
+    t.end();
+  });
+
+  test('valueOf', function(t) {
+    const obj = generator();
+    assertFunction(t, 'valueOf', { obj, expected: obj });
+    t.end();
+  });
+
+  test('__defineGetter__', function(t) {
+    t.plan(16);
+    assertFunction(t, '__defineGetter__', {
+      generator,
+      args: ['foo', () => 'works!'],
+      expected: undefined,
+      afterCall: obj => {
+        t.equal(obj.foo, 'works!', 'should return value when accessing getter property');
+      },
+    });
+  });
+
+  test('__defineSetter__', function(t) {
+    t.plan(16);
+    assertFunction(t, '__defineSetter__', {
+      generator,
+      args: ['foo', x => t.equal(x, 'works!', 'should call setter with set value')],
+      expected: undefined,
+      afterCall: obj => {
+        obj.foo = 'works!';
+      },
+    });
+  });
+
+  test('__lookupGetter__', function(t) {
+    t.plan(11);
+
+    const obj = generator();
+    obj.__defineGetter__('foo', () => 'works!');
+
+    assertFunction(t, '__lookupGetter__', {
+      obj,
+      args: ['foo'],
+      assert: (getter, msg) => t.equal(getter(), 'works!', msg + '()'),
+    });
+  });
+
+  test('__lookupSetter__', function(t) {
+    t.plan(11);
+
+    const obj = generator();
+    let msg;
+
+    obj.__defineSetter__('foo', x => {
+      t.equal(x, 'works!', msg);
+    });
+
+    assertFunction(t, '__lookupSetter__', {
+      obj,
+      args: ['foo'],
+      assert: (setter, _msg) => {
+        msg = _msg + '()';
+        setter('works!');
+      },
+    });
+  });
+});
+
+test('assign', function(t) {
+  const obj1 = { a: 1, b: 2 };
+  const obj2 = { a: 2, c: 3 };
+  const obj3 = assign(obj1, obj2);
+  obj1.b = 42;
+  obj2.c = 42;
+  t.equal(obj3.a, 2);
+  t.equal(obj3.b, 2);
+  t.equal(obj3.c, 3);
+  obj3.a = 42;
+  t.equal(obj1.a, 1);
+  t.equal(obj2.a, 2);
+  t.end();
+});
+
+function assertFunction(t, name, opts) {
+  assertExecution(t, name, opts);
+  assertNonExecutingFunctionPrototypeFunctions(t, name, opts);
+}
+
+function assertNonExecutingFunctionPrototypeFunctions(
+  t,
+  name,
+  { obj: _obj, generator = () => _obj }
+) {
+  const obj = generator();
+
+  // TODO: This is only the case in script-mode. How do we support and test non-script mode?
```

Comment:
@kobelb After moving this module into the Kibana repo, the `'use strict'` comment in the top has been removed by our linter. So now the tests are not run in strict mode and so these assertions fail.

Should we override the linter and enforce strict mode, or update the tests to work in sloppy mode instead? 🤔

Reply:
Fixed